core/src/main/java/org/elasticsearch/index/query/AndQueryBuilder.java

```diff
@@ -127,14 +127,7 @@ public int hashCode() {
     }
 
     @Override
-    public boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        }
-        if (obj == null || getClass() != obj.getClass()) {
-            return false;
-        }
-        AndQueryBuilder other = (AndQueryBuilder) obj;
+    public boolean equals(AndQueryBuilder other) {
```

Comment:
agreed, I feel the same, I originally called it doEquals but wasnt sure about it. Do you have a preference?